ddl/ddl_api.go

```diff
@@ -283,14 +283,18 @@ func buildColumnsAndConstraints(ctx sessionctx.Context, colDefs []*ast.ColumnDef
 }
 
 // ResolveCharsetCollation will resolve the charset by the order: table charset > database charset > server default charset.
-func ResolveCharsetCollation(tblCharset, dbCharset string) (string, string, error) {
+func ResolveCharsetCollation(tblCharset, tblCollate, dbCharset string) (string, string, error) {
 	if len(tblCharset) != 0 {
-		defCollate, err := charset.GetDefaultCollation(tblCharset)
-		if err != nil {
-			// return terror is better.
-			return "", "", ErrUnknownCharacterSet.GenWithStackByArgs(tblCharset)
+		// tblCollate is not specified by user.
+		if len(tblCollate) == 0 {
+			defCollate, err := charset.GetDefaultCollation(tblCharset)
+			if err != nil {
```

Comment:
This function is very confusing. The name is `ResolveCharsetCollation`, and its comment `ResolveCharsetCollation will resolve the charset by the order: table charset > database charset > server default charset.`. I think it would  be better to add comment about `Collation` . Besides, is `dbCollate` exist?

Reply:
Sounds reasonable, it's better to add `dbCollate` as well.

Reply:
Pre-code is not every considerable，just considered the charset problem.  
Addressed, done.